src/api/middleware/utils.js

```diff
@@ -0,0 +1,56 @@
+const HTTPStatus = require("http-status-codes");
+const { APIError, ErrorTypes } = require("./errorHandler");
+const ValidationReasons = require("./validators/validationReasons");
+const lodash = require("lodash");
+
+const DEFAULT_ERROR_CODE = ErrorTypes.VALIDATION_ERROR;
+const DEFAULT_ERROR_MSG = ValidationReasons.UNKNOWN;
+const DEFAULT_STATUS_CODE = HTTPStatus.BAD_REQUEST;
+
+/**
+ * Comibnes array of middleware using OR logic. Only fails if ALL functions fail (either by throwing or calling next(error))
+ *
+ * Each middleware will receive a different req object, no not rely on it to be shared among them
+ *
+ * @param {Function[]} Array of express middleware to be run
+ * @param {object} Options:
+ *  - error_code: error code in case of error (default: ErrorTypes.VALIDATION_ERROR)
+ *  - msg: the message in case of error (default: ValidationReasons.UNKNOWN)
+ *  - status_code: The status used in the HTTP Response in case of error (default: BAD_REQUEST (400))
+ */
+const or = (
+    [...middlewares],
+    {
+        error_code = DEFAULT_ERROR_CODE,
+        msg = DEFAULT_ERROR_MSG,
+        status_code = DEFAULT_STATUS_CODE
+    } = {}
```

Comment:
Why use here this method instead of `{error_code = DEFAULT, ...params}` like in offer creations? We could also change offer creation to this, if this is better

Reply:
I'd say they are equivalent. However, in generateTestOffer, the return is an object base on the input directly, so we can save some lines. Here, a clearer separation between arguments and function body is more appropriate, I think, and we would not save any lines by choosing another approach, since we'd have to do the restructuring inside the function body